packages/libs/client/src/v2/pact.ts

```diff
@@ -0,0 +1,238 @@
+/* eslint-disable @kadena-dev/no-eslint-disable */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import { createExp } from '@kadena/pactjs';
+
+import { parseType } from '../utils/parseType';
+
+const prop = <T extends any>(name: string, value?: T) =>
+  value === undefined ? {} : { [name]: value };
+
+export const getModule = (name: string) => {
+  let code = name;
+  const pr: any = new Proxy<any>(function () {} as unknown, {
+    get(_, path: string) {
+      code += '.' + path;
+      return pr;
+    },
+    apply(_, thisArg, args) {
+      const exp = createExp(code, ...args.map(parseType));
+      code = name;
+      return exp;
+    },
+  });
+  return pr;
+};
+
+interface IPayload {
+  exec: <
+    T extends Array<{
+      capability(name: string, ...args: any): ICapabilityItem;
+    }>,
+  >(
+    codes: T,
+    data?: any,
+    // use _branch to add type inferring for using it when user call signer function then we can show a related list of capabilities
+  ) => { payload: IExecPayload & { funs: [...T]; _brand: 'exec' } };
+  cont: (options: IContinuationPayload) => {
+    payload: IContinuationPayload & { _brand: 'cont' };
+  };
+}
+
+export const payload: IPayload = {
+  exec: (codes, data) => ({
+    payload: {
+      code: codes.join(''),
+      ...prop('data', data),
+    } as any,
+  }),
+  cont: (options) => ({
+    payload: options as any,
+  }),
+};
+
+export type IPartialCommand = Partial<Omit<ICommand, 'payload'>>;
+
+// TODO : improve the return value to merge all of the inputs as an object
+interface ICommandBuilder {
+  <F extends Pick<ICommand, 'payload'>>(
+    payload: F,
+    ...second: [
+      ...Array<
+        | Partial<ICommand>
+        | ((payload: F & Partial<ICommand>) => Partial<ICommand>)
+      >,
+    ]
+  ): { command: Partial<ICommand>; stringify: () => string };
+}
+
+export const commandBuilder: ICommandBuilder = (first: any, ...rest: any) => {
+  const args: Array<
+    Partial<ICommand> | ((cmd: Partial<ICommand>) => Partial<ICommand>)
+  > = [first, ...rest] as any;
+  const command = args.reduce<Partial<ICommand>>((acc, item) => {
+    const part = typeof item === 'function' ? item(acc) : item;
+    if (part.payload !== undefined) {
```

Comment:
Should this logic be here in the command builder? I'd expect this logic to be present at the reducers that would modify the `payload`. Then this function can stay dumb. (same for all the other data checks below)

Reply:
yes I removed it